### PR TITLE
fix: broken attach and attach another button for searchable associations

### LINKED ISSUE #3456: `Attach and Attach Another` action refreshing entire page when `searchable: true` on `has_many` field
### Steps to Reproduce
1. `git clone git@github.com:avo-hq/main.avodemo.com.git`
2. in `app/avo/resources/project.rb`:
```diff
-    field :users, as: :has_and_belongs_to_many
+    field :users, as: :has_and_belongs_to_many, searchable: true
```
3. Attempt to `Attach and Attach Another` User on the Projects resource

### Expected behavior & Actual behavior
Expected: `Attach and Attach Another` keeps the modal open so you can attach another record
Actual: `Attach and Attach Another` refreshes the whole page, closing the modal.

### Models and resource files
in `app/avo/resources/project.rb`:
```diff
-    field :users, as: :has_and_belongs_to_many
+    field :users, as: :has_and_belongs_to_many, searchable: true
```

### System configuration
**Avo version**: 13.4.1

**Rails version**: 7.2.0beta2 / 8.0.0

**Ruby version**: 3.3.0 / 3.3.4

**License type**:

 - [ ] Community
 - [ ] Pro
 - [x] Advanced

**Are you using Avo monkey patches, overriding views or view components?**

 - [ ] Yes. If so, please post code samples.
 - [x] No

### Screenshots or screen recordings
https://github.com/user-attachments/assets/23547ff8-8622-486f-b543-0cea180f1aac

### Additional context
Probably unrelated, but seems to be a bug in the `search_controller.rb`
```javascript
Uncaught (in promise) TypeError: Cannot read properties of null (reading 'remove')
    at n.handleOnSelect (search_controller.js:223:51)
    at Object.onSelect (getDefaultProps.js:90:22)
    at getPropGetters.js:312:18
    at createCancelablePromise.js:60:12
```

#### Impact
 - [ ] High impact (It makes my app un-usable.)
 - [x] Medium impact (I'm annoyed, but I'll live.)
 - [ ] Low impact (It's really a tiny thing that I could live with.)

#### Urgency

 - [ ] High urgency (I can't continue development without it.)
 - [ ] Medium urgency (I found a workaround, but I'd love to have it fixed.)
 - [x] Low urgency (It can wait. I just wanted you to know about it.)

### PR DESCRIPTION
**Pull request by ObiWanKeoni**
# Description

I _think_ this is all that's necessary to fix the issue linked below. In the AssociationsController, we expect a `:turbo_frame` param to be passed to the route if we want to rerender the turbo frame instead of closing it. The only bit I'm unclear on is whether we need to have additional special handling for the `attach_another` button. I figured [this](https://github.com/avo-hq/avo/blob/f6d132891fdf87a754af0880a3a97b9b3e0d4f51/app/controllers/avo/associations_controller.rb#L256) was enough.

Fixes #3456 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Manual review steps

1. Step 1
1. Step 2

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

**Review by Paul-Bob — CHANGES_REQUESTED** [review]
Hi @ObiWanKeoni thank you for looking into this and opening this PR!

What happens is that [this hidden form field](https://github.com/avo-hq/avo/blob/main/app/views/avo/associations/new.html.erb#L44) was introduced only for **non**-searchable fields.

To complete this PR let's remove that hidden field and keep the turbo frame on the url params, **or** remove it from the params and move the hidden form field outside the if searchable condition ensuring that it always  render.

File: app/views/avo/associations/new.html.erb [hunk 1]
```diff
@@ -41,8 +41,14 @@
...
                 classes: 'w-full'
               %>
-              <%= hidden_field_tag :turbo_frame, params[:turbo_frame] %>
             <% end %>
+            <%#
+              It is important to render this hidden field to allow selective rendering of the turbo frame.
+              (e.g. in the case of "Attach and Attach Another")
+
+              Without this, the entire page will be reloaded when the form is submitted.
+            %>
```
Comment (ObiWanKeoni) [inline app/views/avo/associations/new.html.erb:50]: @Paul-Bob I added a comment here for communication purposes. LMK if it should be reworded or if this works. Thank you for your comment - I completely missed the `hidden_field_tag`!

Comment (Paul-Bob) [reply app/views/avo/associations/new.html.erb:50]: Awesome! Thank you @ObiWanKeoni!

**Review by Paul-Bob — APPROVED** [review]
Thanks for this contribution @ObiWanKeoni!

**Pull request merged**